src/CrackingCodeInterview/1.2.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Check if a given string is a permutation of a second one
+ * - Time complexity: O(n+n) ~ O(n)
+ * @param {string} strOne - Main string
+ * @param {string} strTwo - Permutation of the main string
+ * @returns {boolean} Return false if the two given strings are not permutation between them
+ */
+export default function checkPermutationOptimized(strOne, strTwo) {
+  if (strOne.length !== strTwo.length) return false;
+
+  const strOneCounter = {};
+
+  for (let i = 0; i < strOne.length; i += 1) {
+    const char = strOne.charAt(i);
+    strOneCounter[char] = (strOneCounter[char] || 0) + 1;
+  }
+
+  for (let i = 0; i < strTwo.length; i += 1) {
+    strOneCounter[strTwo.charAt(i)] -= 1;
```

Comment:
I think you can use bracket notation to simplify code :question:

Reply:
same of above, done!